agent/structs/config_entry_gateways.go

```diff
@@ -186,6 +198,11 @@ func (e *IngressGatewayConfigEntry) Validate() error {
 			if s.NamespaceOrDefault() == WildcardSpecifier {
 				return fmt.Errorf("Wildcard namespace is not supported for ingress services (listener on port %d)", listener.Port)
 			}
+			sid := NewServiceID(s.Name, &s.EnterpriseMeta)
+			if _, ok := serviceNames[sid]; ok {
+				return fmt.Errorf("Service %s cannot be added multiple times (listener on port %d)", sid, listener.Port)
+			}
+			serviceNames[sid] = struct{}{}
```

Comment:
This is _technically_ a breaking change in the sense that we didn't error before if you had duplicate services in an ingress gateway config so it "worked" although it's not clear what it actually means.

I _think_ I'm OK with calling it a bug though rather than a feature - it seems like impact would be extremely low and very easy to mitigate, nothing would break initially, just config entries that validated before could possibly fail after this but can be fixes without any behaviour change.

Does that seem reasonable for a point release? I think it's OK.

The reason it now actually matters is that you could configure different header manip on two duplicate services with the same service name and it becomes ambiguous which rules should apply to traffic for that service.

We _could_ instead not make this an error but just document that if there duplicates we'll always ignore them after the first one or something?